### PR TITLE
use .counts() rather than deprecated .stats() for GC3Pie

### PR DESCRIPTION
**Pull request by boegel**
@riccardomurri Please review?

### CODE REVIEW

File: easybuild/tools/job/gc3pie.py [hunk 1]
```diff
@@ -220,14 +220,8 @@ def complete(self):
...
-        # `Engine.stats()` (which is used later on in `_print_status_report()`)
-        # changed between 2.4.2 and 2.5.0.dev -- make sure we stay compatible
-        # with both
-        try:
-            self._engine.init_stats_for(Application)
-        except AttributeError:
-            _log.debug("No `init_stats_for` method in the Engine class;"
-                       " assuming pre-2.5.0 GC3Pie and ignoring error.")
+        # prepare for .counts() in _print_status_report
+        self._engine.init_counts_for(Application)
```
Comment (riccardomurri) [inline easybuild/tools/job/gc3pie.py]: This is actually no longer required: counts for `Application` are init'ed by default in the `Engine` class. (This changed during the development of 2.5)

Comment (riccardomurri) [reply easybuild/tools/job/gc3pie.py]: (Does no harm to keep -- it's just redundant now.)

Comment (boegel) [reply easybuild/tools/job/gc3pie.py]: Probably better to just drop it then?

Comment (riccardomurri) [reply easybuild/tools/job/gc3pie.py]: If the minimally required version of GC3Pie is 2.5.0, then yes.

Comment (boegel) [reply easybuild/tools/job/gc3pie.py]: That's indeed the case since #2554, so I'll drop it. Thanks for the feedback!

**Pull request merged**